### PR TITLE
{phys}[foss/2022a] Delft3D v4.04.01

### PR DESCRIPTION
**Pull request by WilleBell**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by MaximeVdB** [comment]
Hi Cintia -- I just wanted to mention this easyconfig that we wrote for installing Delft3D, in case you hadn't seen it yet:

https://github.com/vscentrum/vsc-software-stack/tree/site-kul/d/Delft3D

It's a different toolchain and different Delft3D version, so I don't know how useful it would be (its additional sanity checks might be worth considering though).

**Comment by WilleBell** [comment]
> Hi Cintia -- I just wanted to mention this easyconfig that we wrote for installing Delft3D, in case you hadn't seen it yet:
> 
> https://github.com/vscentrum/vsc-software-stack/tree/site-kul/d/Delft3D
> 
> It's a different toolchain and different Delft3D version, so I don't know how useful it would be (its additional sanity checks might be worth considering though).

Thank you for this Maxime, it was useful to see how you could install Delft in a different toolchain. This PR is actually only for the FLOW module, that's why there are less sanity checks.

**Review by jfgrimm — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/d/Delft3D-FLOW/Delft3D-FLOW-4.04.01-foss-2022a.eb [hunk 1]
```diff
@@ -0,0 +1,71 @@
+easyblock = 'ConfigureMake'
+
+name = 'Delft3D-FLOW'
```
[the remark this answers is not in the document]
Comment (WilleBell) [reply easybuild/easyconfigs/d/Delft3D-FLOW/Delft3D-FLOW-4.04.01-foss-2022a.eb]: Delft3D has different modules, this installation is only for the FLOW module. I believe the "FLOW" reference is needed, would you agree on a version suffix?

[the remark this answers is not in the document]
Comment (jfgrimm) [reply easybuild/easyconfigs/d/Delft3D-FLOW/Delft3D-FLOW-4.04.01-foss-2022a.eb]: versionsuffix makes sense, thanks :)

**Comment by jfgrimm** [comment]
@boegelbot: please test @ generoso

**Comment by jfgrimm** [comment]
forgot to put the source in place first :facepalm:

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**FAILED**
Build succeeded for 4 out of 7 (1 easyconfigs in total)
node114.viking2.yor.alces.network - Linux Rocky Linux 8.8, x86_64, AMD EPYC 7643 48-Core Processor, Python 3.6.8
See https://gist.github.com/jfgrimm/aedb745a244a9fdb5f274bfeee87f2cb for a full test report.

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**FAILED**
Build succeeded for 2 out of 3 (1 easyconfigs in total)
node114.viking2.yor.alces.network - Linux Rocky Linux 8.8, x86_64, AMD EPYC 7643 48-Core Processor, Python 3.6.8
See https://gist.github.com/jfgrimm/dc3929048b888883806eb53f3f67e063 for a full test report.

**Comment by jfgrimm** [comment]
@smoors yes, I've noticed that. I'm trying out a patch that removes the bundled esmf stuff atm

File: easybuild/easyconfigs/d/Delft3D/Delft3D-4.04.01-foss-2022a-FLOW.eb [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    ('Autotools', '20220317'),
+    ('M4', '1.4.19'),
+]
+
+dependencies = [
+    ('netCDF-Fortran', '4.6.0'),
+    ('METIS', '5.1.0'),
+    ('PETSc', '3.17.4'),
+    ('expat', '2.4.8'),
+    ('ESMF', '8.3.0'),
```
Comment (smoors) [inline easybuild/easyconfigs/d/Delft3D/Delft3D-4.04.01-foss-2022a-FLOW.eb:47]: ESMF is not required, can be removed as none of the binaries or libraries depend on it.
i guess it is there for the other Delft3D components.

Comment (boegel) [reply easybuild/easyconfigs/d/Delft3D/Delft3D-4.04.01-foss-2022a-FLOW.eb:47]: @smoors No longer relevant thanks to the changes made via https://github.com/WilleBell/easybuild-easyconfigs/pull/3?

**Comment by boegel** [comment]
@boegelbot: please test @ generoso

File: easybuild/easyconfigs/d/Delft3D/Delft3D-4.04.01-foss-2022a-FLOW.eb (hunk 2 above)
Comment (smoors) [reply easybuild/easyconfigs/d/Delft3D/Delft3D-4.04.01-foss-2022a-FLOW.eb:47]: it's the nice fix of @jfgrimm  in https://github.com/WilleBell/easybuild-easyconfigs/pull/3 that introduced this dependency, which i think is not necessary.

Comment (jfgrimm) [reply easybuild/easyconfigs/d/Delft3D/Delft3D-4.04.01-foss-2022a-FLOW.eb:47]: no idea whether that would be part of the workflow, but it did get installed. I'll have a look at patching it to use our ESMF dep

Comment (smoors) [reply easybuild/easyconfigs/d/Delft3D/Delft3D-4.04.01-foss-2022a-FLOW.eb:47]: hm indeed, better to keep it in then

Comment (jfgrimm) [reply easybuild/easyconfigs/d/Delft3D/Delft3D-4.04.01-foss-2022a-FLOW.eb:47]: @smoors @WilleBell should be fixed in https://github.com/WilleBell/easybuild-easyconfigs/pull/4

**Comment by boegel** [comment]
Awaiting merge of https://github.com/WilleBell/easybuild-easyconfigs/pull/4 so missing script is also installed

**Review by jfgrimm — APPROVED** [review]
lgtm

**Comment by jfgrimm** [comment]
@boegelbot please test @ generoso

**Comment by jfgrimm** [comment]
Going in, thanks @WilleBell!

**Pull request merged**